### DISCUSSION TITLE
Using ENUM in nested arguments

### DISCUSSION THREAD

**Discussion by GuiuGaldino** [IlyaMatsuev/Apex-GraphQL-Client#33] [Q&A] [answered]
Hello, I am trying to use enum in a nested argument in a mutation request but it seens it is not working properly. See the example below and the  mutation when i run revokeDrivePrevileges .build():
```
GraphQLField revokeDrivePrevileges = new GraphQLField('revokeCoReader')
            .withArgument('brandId', 'BMW')
            .withArgument('carType', 'SUV')
            .withArgument('accountId', '12345768')
            .withArgument(
                'metadata',
                new Map<String, Object>{
                    'channel' => new GraphQLEnum(Constants.CHANNEL),
                    'authorType' => new GraphQLEnum(Constants.AUTHOR_TYPE),
                    'agentId' => '123456'
                }
            );

```  
`revokeCoReader(brandId:"BMW",carType:"SUV",accountId:"12345768",metadata:{agentId:"123456",authorType:{value:"AGENT"},channel:{value:"SALESFORCE"}})`

I am doing something wrong? Is there a workaround that I can use?

**Comment by IlyaMatsuev** [comment]
Hi @GuiuGaldino!
At first, seems like an issue of serializing graphql enums into proper string values when they're used in a map like this. I will take a look into it (today/tomorrow) and see if I can do something to make It work as expected.

Unfortunately, I can't think of any workarounds here for now apart from changing the structure of your request so that your enum values are separate arguments.

**Comment by IlyaMatsuev** [comment, marked as answer]
Hey @GuiuGaldino, I have just published a new version of the package with the fix for your problem, here's issue #34 with PR attached. Please let me know if this helped!

**Comment by GuiuGaldino** [comment]
It is working as expected. Thanks for the help!!
